Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -0,0 +1,2031 @@
+            continue;
+        }
+
+        auto commandDictionary = commandsDictionary->getObject(commandIdentifier);
+        if (!commandDictionary->size()) {
+            error = WEB_UI_STRING("Empty or invalid command in the `commands` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for invalid command");
+            continue;
+        }
+
+        CommandData commandData;
+        commandData.identifier = commandIdentifier;
+        commandData.activationKey = emptyString();
+        commandData.modifierFlags = { };
+
+        bool isActionCommand = false;
+        if (supportsManifestVersion(3) && commandData.identifier == "_execute_action"_s)
+            isActionCommand = true;
+        else if (!supportsManifestVersion(3) && (commandData.identifier == "_execute_browser_action"_s || commandData.identifier == "_execute_page_action"_s))
+            isActionCommand = true;
+
+        if (isActionCommand && !hasActionCommand)
+            hasActionCommand = true;
+
+        // Descriptions are required for standard commands, but are optional for action commands.
+        auto description = commandDictionary->getString(commandsDescriptionKeyManifestKey);
+        if (description.isEmpty() && !isActionCommand) {
+            error = WEB_UI_STRING("Empty or invalid `description` in the `commands` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for invalid command description");
+            continue;
+        }
+
+        if (isActionCommand && !description.isEmpty()) {
+            description = displayActionLabel();
+            if (!description.isEmpty())
+                description = displayShortName();
+        }
+
+        commandData.description = description;
+
+        if (auto suggestedKeyDictionary = commandDictionary->getObject(commandsSuggestedKeyManifestKey)) {
+#if PLATFORM(MAC) || PLATFORM(IOS_FAMILY)
+            const auto macPlatform = "mac"_s;
+            const auto iosPlatform = "ios"_s;
+#elif PLATFORM(GTK)
+            const auto linuxPlatform = "linux"_s;
+#endif
+            const auto defaultPlatform = "default"_s;
+
+#if PLATFORM(MAC)
+            auto platformShortcut = !suggestedKeyDictionary->getString(macPlatform).isEmpty() ? suggestedKeyDictionary->getString(macPlatform) : suggestedKeyDictionary->getString(iosPlatform);
+#elif PLATFORM(IOS_FAMILY)
+            auto platformShortcut = !suggestedKeyDictionary->getString(iosPlatform).isEmpty() ? suggestedKeyDictionary->getString(iosPlatform) : suggestedKeyDictionary->getString(macPlatform);
+#elif PLATFORM(GTK)  
+            auto platformShortcut = suggestedKeyDictionary->getString(linuxPlatform);
+#else
+            auto platformShortcut = suggestedKeyDictionary->getString(defaultPlatform);
+#endif
+            if (platformShortcut.isEmpty())
+                platformShortcut = suggestedKeyDictionary->getString(defaultPlatform);
```

Comment:
Can return `nullptr`s.

Reply:
getString() doesn't return a nullptr. It can return a nullString though, but that's still a String object. inside parseCommandShortcut (what the platformSHortcut goes to, we check for an empty string and handle it already.